File: treebolicIface/src/main/java/org/treebolic/TreebolicIface.java
/*
 * Copyright (c) 2019-2023. Bernard Bou
 */

package org.treebolic;

/**
 * Constants used in Android
 * TODO remove from here
 */
public interface TreebolicIface
{
	/**
	 * Whether to use model references
	 */
	boolean USE_MODEL_REFERENCES = false;

	// intent extras

	/**
	 * Make model
	 */
	String ARG_SOURCE = "org.treebolic.SOURCE";

	/**
	 * Plugin package
	 */
	String ARG_PLUGINPKG = "org.treebolic.PLUGINPKG";

	/**
	 * Provider
	 */
	String ARG_PROVIDER = "org.treebolic.PROVIDER";

	/**
	 * Service
	 */
	String ARG_SERVICE = "org.treebolic.CLIENT"; // TODO

	/**
	 * Base
	 */
	String ARG_BASE = "org.treebolic.BASE";

	/**
	 * Image base
	 */
	String ARG_IMAGEBASE = "org.treebolic.IMAGEBASE";

	/**
	 * Model reference
	 */
	String ARG_MODEL_REFERENCE = "org.treebolic.MODEL_REFERENCE";

	/**
	 * Model
	 */
	String ARG_MODEL = "org.treebolic.MODEL";

	/**
	 * Serialized
	 */
	String ARG_SERIALIZED = "org.treebolic.SERIALIZED";

	/**
	 * Serialized model URI
	 */
	String ARG_SERIALIZED_MODEL_URI = "org.treebolic.SERIALIZED_MODEL_URI";

	/**
	 * Settings
	 */
	String ARG_SETTINGS = "org.treebolic.SETTINGS";

	/**
	 * Style
	 */
	String ARG_STYLE = "org.treebolic.STYLE";

	/**
	 * URL scheme
	 */
	String ARG_URLSCHEME = "org.treebolic.URLSCHEME";

	/**
	 * More
	 */
	String ARG_MORE = "org.treebolic.MORE";

	/**
	 * Parent Activity
	 */
	String ARG_PARENTACTIVITY = "org.treebolic.PARENT";

	// action

	/**
	 * Make model
	 */
	String ACTION_MAKEMODEL = "org.treebolic.action.MAKE_MODEL";

	// activities

	/**
	 * Treebolic package
	 */
	String PKG_TREEBOLIC = "org.treebolic";

	/**
	 * Treebolic activity
	 */
	String ACTIVITY_TREEBOLIC = "org.treebolic.TreebolicActivity";

	/**
	 * Treebolic plugin
	 */
	String ACTIVITY_PLUGIN = "org.treebolic.TreebolicPluginActivity";

	/**
	 * Treebolic model
	 */
	String ACTIVITY_MODEL = "org.treebolic.TreebolicModelActivity";

	/**
	 * Treebolic client
	 */
	String ACTIVITY_CLIENT = "org.treebolic.TreebolicClientActivity";

	// preference keys

	/**
	 * Source preference key
	 */
	String PREF_SOURCE = "pref_source";

	/**
	 * Base preference key
	 */
	String PREF_BASE = "pref_base";

	/**
	 * Image base preference key
	 */
	String PREF_IMAGEBASE = "pref_imagebase";

	/**
	 * Settings preference key
	 */
	String PREF_SETTINGS = "pref_settings";
}

File: treebolicIface/src/main/java/org/treebolic/TreebolicIface.kt
/*
 * Copyright (c) 2019-2023. Bernard Bou
 */
package org.treebolic

/**
 * Constants used in Android
 */
interface TreebolicIface {

    companion object {

        /**
         * Whether to use model references
         */
        const val USE_MODEL_REFERENCES: Boolean = false

        // intent extras

        /**
         * Make model
         */
        const val ARG_SOURCE: String = "org.treebolic.SOURCE"

        /**
         * Plugin package
         */
        const val ARG_PLUGINPKG: String = "org.treebolic.PLUGINPKG"

        /**
         * Provider
         */
        const val ARG_PROVIDER: String = "org.treebolic.PROVIDER"

        /**
         * Service
         */
        const val ARG_SERVICE: String = "org.treebolic.SERVICE"

        /**
         * Base
         */
        const val ARG_BASE: String = "org.treebolic.BASE"

        /**
         * Image base
         */
        const val ARG_IMAGEBASE: String = "org.treebolic.IMAGEBASE"

        /**
         * Model reference
         */
        const val ARG_MODEL_REFERENCE: String = "org.treebolic.MODEL_REFERENCE"

        /**
         * Model
         */
        const val ARG_MODEL: String = "org.treebolic.MODEL"

        /**
         * Serialized
         */
        const val ARG_SERIALIZED: String = "org.treebolic.SERIALIZED"

        /**
         * Serialized model URI
         */
        const val ARG_SERIALIZED_MODEL_URI: String = "org.treebolic.SERIALIZED_MODEL_URI"

        /**
         * Settings
         */
        const val ARG_SETTINGS: String = "org.treebolic.SETTINGS"

        /**
         * Style
         */
        const val ARG_STYLE: String = "org.treebolic.STYLE"

        /**
         * URL scheme
         */
        const val ARG_URLSCHEME: String = "org.treebolic.URLSCHEME"

        /**
         * More
         */
        const val ARG_MORE: String = "org.treebolic.MORE"

        /**
         * Parent Activity
         */
        const val ARG_PARENTACTIVITY: String = "org.treebolic.PARENT"

        // action
        /**
         * Make model
         */
        const val ACTION_MAKEMODEL: String = "org.treebolic.action.MAKE_MODEL"

        // activities

        /**
         * Treebolic package
         */
        const val PKG_TREEBOLIC: String = "org.treebolic"

        /**
         * Treebolic activity
         */
        const val ACTIVITY_TREEBOLIC: String = "org.treebolic.TreebolicActivity"

        /**
         * Treebolic plugin
         */
        const val ACTIVITY_PLUGIN: String = "org.treebolic.TreebolicPluginActivity"

        /**
         * Treebolic model
         */
        const val ACTIVITY_MODEL: String = "org.treebolic.TreebolicModelActivity"

        /**
         * Treebolic client
         */
        const val ACTIVITY_CLIENT: String = "org.treebolic.TreebolicClientActivity"

        // preference keys

        /**
         * Source preference key
         */
        const val PREF_SOURCE: String = "pref_source"

        /**
         * Base preference key
         */
        const val PREF_BASE: String = "pref_base"

        /**
         * Image base preference key
         */
        const val PREF_IMAGEBASE: String = "pref_imagebase"

        /**
         * Settings preference key
         */
        const val PREF_SETTINGS: String = "pref_settings"
    }
}
